### DISCUSSION TITLE
Connect to xvfb screen using VNC

### DISCUSSION THREAD

**Discussion by stvenchg** [seleniumbase/SeleniumBase#3254] [Q&A] [answered]
Hi guys,

I'm trying to automate a click on an element using PyAutoGUI in an xvfb environment. However, the click isn't being registered. I'd like to connect to the xvfb instance to observe where the cursor is moving. Is this possible? Thank you.

**Comment by mdmintz** [comment, marked as answer]
From Google AI:

<img width="1000" alt="Screenshot 2024-11-11 at 3 29 57 PM" src="https://github.com/user-attachments/assets/3876da8f-21c1-4cf2-ab43-90e8458adcf1">

------

### Using PyAutoGUI with VNC can be tricky due to a few key considerations:
**Challenges:**
* **Coordinate Mapping:** The coordinates on your local machine might not match the coordinates on the remote VNC machine.
* **Window Focus:** PyAutoGUI relies on the active window. If the VNC window loses focus, your automation might fail.
* **Performance:** VNC can introduce lag, affecting the timing of your automation scripts.

------

Solution: Don't use VNC if you need to use PyAutoGUI actions.
